src/emqx_listeners.erl

```diff
@@ -173,9 +173,15 @@ with_port({Addr, Port}, Opts = #{socket_opts := SocketOption}) ->
 restart() ->
     lists:foreach(fun restart_listener/1, emqx:get_env(listeners, [])).
 
--spec(restart_listener(listener()) -> any()).
+% emqx_listeners:find_by_id(Identifier)
+-spec(restart_listener(listener() | string() | binary()) -> any()).
```

Comment:
return spec:
`ok | {error, any()}`

Reply:
the return spec for `restart_listener/3` should be fixed too.

Reply:
ah, sorry, no, we call `restart_listener/1` here, which returns `ok | {error, any()}`